e2e/test/IoTHubServiceProxyE2ETests.cs

```diff
@@ -89,17 +92,30 @@ private async Task RegistryManager_AddDevice(HttpTransportSettings httpTransport
 
         private async Task JobClient_ScheduleAndRunTwinJob(HttpTransportSettings httpTransportSettings)
         {
-            string jobId = "JOBSAMPLE" + Guid.NewGuid().ToString();
-            string query = $"DeviceId IN ['{JobDeviceId}']";
-
             Twin twin = new Twin(JobDeviceId);
             twin.Tags = new TwinCollection();
             twin.Tags[JobTestTagName] = JobDeviceId;
 
             using (JobClient jobClient = JobClient.CreateFromConnectionString(s_connectionString, httpTransportSettings))
             {
-                JobResponse createJobResponse = await jobClient.ScheduleTwinUpdateAsync(jobId, query, twin, DateTime.UtcNow, (long)TimeSpan.FromMinutes(2).TotalSeconds).ConfigureAwait(false);
-                JobResponse jobResponse = await jobClient.GetJobAsync(jobId).ConfigureAwait(false);
+                int tryCount = 0;
+                while (true)
+                {
+                    try
+                    {
+                        string jobId = "JOBSAMPLE" + Guid.NewGuid().ToString();
+                        string query = $"DeviceId IN ['{JobDeviceId}']";
+                        JobResponse createJobResponse = await jobClient.ScheduleTwinUpdateAsync(jobId, query, twin, DateTime.UtcNow, (long)TimeSpan.FromMinutes(2).TotalSeconds).ConfigureAwait(false);
+                        break;
+                    }
+                    // Concurrent jobs can be rejected, so implement a retry mechanism to handle conflicts with other tests
+                    catch (ThrottlingException) when (++tryCount < MaxIterationWait)
+                    {
+                        _log.WriteLine($"ThrottlingException... waiting.");
+                        await Task.Delay(_waitDuration).ConfigureAwait(false);
```

Comment:
so basically with adding a delay of 5 second before retry, we do not see throttling exception?

Reply:
No it would probably take more than that. About 30-40 seconds. We will have about 5 jobs running in parallel so depends on the contention.